### PR TITLE
 [cmake] do not use $ROOTSYS/include when building ROOT executables 

### PR DESCRIPTION
**Pull request by linev**
As all libraries, executabls have to use source directories for build. 
In this case only includes from linked libraries are allowed to be used

Affects all executables and all tests. Should improve building with `-Dcxxmodules=ON`.

Several generated includes moved to "ginclude/" path while used only in ROOT:
* `hist2workspaceCommandLineOptionsHelp.h`
* `haddCommandLineOptionsHelp.h`
* `rootCommandLineOptionsHelp.h`

Several adjustments in tests are required. 

Also in several places replace `include_directories` by `target_include_directories`

### CODE REVIEW

File: cmake/modules/RootMacros.cmake [hunk 1]
```diff
@@ -1338,9 +1338,36 @@ function(ROOT_EXECUTABLE executable)
   if(ARG_TEST) # we are building a test, so add EXCLUDE_FROM_ALL
     set(_all EXCLUDE_FROM_ALL)
   endif()
-  include_directories(BEFORE ${CMAKE_BINARY_DIR}/include)
-  add_executable( ${executable} ${_all} ${exe_srcs})
-  target_link_libraries(${executable} ${ARG_LIBRARIES} )
+  if(NOT (PROJECT_NAME STREQUAL "ROOT"))
+    # only for non-ROOT executable use $ROOTSYS/include
+    include_directories(BEFORE ${CMAKE_BINARY_DIR}/include)
```
Comment (Axel-Naumann) [inline cmake/modules/RootMacros.cmake:1343]: But if not `ROOT`, is `CMAKE_BINARY_DIR` really `ROOTSYS`?

Comment (linev) [reply cmake/modules/RootMacros.cmake:1343]: No idea, all logic around `(PROJECT_NAME STREQUAL "ROOT")` not always clear for me.
But again - idea of my changes - adjust code only for ROOT itself

File: core/rootcling_stage1/CMakeLists.txt [hunk 2]
```diff
@@ -37,5 +35,12 @@ ROOT_EXECUTABLE(rootcling_stage1 src/rootcling_stage1.cxx
                               $<TARGET_OBJECTS:Foundation_Stage1>
                               LIBRARIES ${CLING_LIBRARIES} ${LINK_LIBS} ${CMAKE_DL_LIBS} ${CMAKE_THREAD_LIBS_INIT}
                               NOINSTALL)
+
+target_include_directories(rootcling_stage1 PRIVATE
+   ${CMAKE_SOURCE_DIR}/core/base/inc    # for RConfig.hxx
+   ${CMAKE_SOURCE_DIR}/core/dictgen/res # for rootcling_impl.h
+   ${CMAKE_BINARY_DIR}/ginclude         # for RConfigure.h
```
Comment (oshadura) [inline core/rootcling_stage1/CMakeLists.txt:42]: @Axel-Naumann I see that we need to prioritize PR.... I tried to build ROOT master (without this PR) from scratch and I see next failure:
`[ 41%] Building CXX object core/rootcling_stage1/CMakeFiles/rootcling_stage1.dir/src/rootcling_stage1.cxx.o
cd /home/oksana/CERN_sources/root-rtcxxmodules/builds/core/rootcling_stage1 && /usr/bin/c++   -I/home/oksana/CERN_sources/root-rtcxxmodules/builds/include -I/home/oksana/CERN_sources/root-rtcxxmodules/core/dictgen/res  -std=c++11 -Wno-implicit-fallthrough -Wno-noexcept-type -pipe  -Wshadow -Wall -W -Woverloaded-virtual -fsigned-char -pthread -O3 -DNDEBUG   -std=c++11 -fno-rtti -o CMakeFiles/rootcling_stage1.dir/src/rootcling_stage1.cxx.o -c /home/oksana/CERN_sources/root-rtcxxmodules/core/rootcling_stage1/src/rootcling_stage1.cxx
/home/oksana/CERN_sources/root-rtcxxmodules/core/rootcling_stage1/src/rootcling_stage1.cxx:12:10: fatal error: RConfigure.h: No such file or directory
   12 | #include "RConfigure.h" `

Comment (oshadura) [reply core/rootcling_stage1/CMakeLists.txt:42]: This PR will solve a current problem...

**Pull request merged**